orchagent/watermarkorch.cpp

```diff
@@ -94,6 +85,52 @@ void WatermarkOrch::doTask(Consumer &consumer)
     }
 }
 
+void WatermarkOrch::handleWmConfigUpdate(const std::string &key, const std::vector<FieldValueTuple> &fvt)
+{
+    if (key == "TELEMETRY_INTERVAL")
+    {
+        for (std::pair<std::basic_string<char>, std::basic_string<char> > i: fvt)
+        {
+            if (i.first == "interval")
+            {
+                auto intervT = timespec { .tv_sec = to_uint<uint32_t>(i.second.c_str()) , .tv_nsec = 0 };
+                m_telemetryTimer->setInterval(intervT);
```

Comment:
I just checked the implementation of setInterval(), https://github.com/Azure/sonic-swss-common/blob/master/common/selectabletimer.cpp#L63. It only changes the value of the itimerspec struct, not sync the value to the timer. If the timer is not armed, it will not start the timer. This is good as we may not want to start the timer, but only want to change the interval. If the timer is armed, we will need a call to timerfd_settime(), which is called by SelectableTimer::start() and SelectableTimer::reset(), to sync the actual change to the timer. This means the change will take effect at a later timer when the current timer expires and the SelectableTimer doTask handler is called. If we do not care about the interval to take immediate effect. This is also fine.

Reply:
The timer is expected to be changed after current interval has ended. This is stated in the design.

Reply:
OK, let's follow the design for now. Then, we need a call to sync change to the actual timer in the doTask(SelectableTimer &timer) if we see there is a change in the telemetry interval